.github/workflows/codeql.yaml

```diff
@@ -0,0 +1,35 @@
+name: "CodeQL Advanced"
+
+on: [pull_request]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  analyze:
+    name: Analyze (${{ matrix.language }})
+    runs-on: ubuntu-latest
+    permissions:
+      security-events: write
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - language: java-kotlin
+            build-mode: none
```

Comment:
Any reason you let that in?

Reply:
What exactly do you mean? The build-mode line? CodeQL uses build mode to decide whether compilation is needed (e.g Kotlin) or not (e.g. for Java). Setting to `none` indicates we don't compile here.